hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -550,6 +595,30 @@ public AbfsRestOperation checkAccess(String path, String rwx)
     return op;
   }
 
+  /**
+   * If configured for SAS AuthType, appends SAS token to queryBuilder
+   * @param path
+   * @param operation
+   * @param queryBuilder
+   * @throws SASTokenProviderException
+   */
+  private void appendSASTokenToQuery(String path, String operation, AbfsUriQueryBuilder queryBuilder) throws SASTokenProviderException {
+    if (this.authType == AuthType.SAS) {
+      try {
+        LOG.trace("Fetch SAS token for {} on {}", operation, path);
```

Comment:
This is a hot path and we are calling LOG.trace twice (L608 and L612), is this expensive?  Could you step thru in debugger to confirm this is a light weight Boolean check only, when the level is not TRACE?

Reply:
log @ debug/trace is low cost as long as you aren't actually concatenating strings; the sole cost is to instantiate the "Fetch SAS token.." string to pass down a reference.

Reply:
Good to know, then we can resolve this comment.  (I don't see an option to resolve comments.)